internal/activity/database/activity_repository.go

```diff
@@ -0,0 +1,119 @@
+package database
+
+import (
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/sumelms/microservice-activity/internal/activity/domain"
+	"github.com/sumelms/microservice-activity/pkg/errors"
+)
+
+// NewActivityRepository creates the activity ActivityRepository.
+func NewActivityRepository(db *sqlx.DB) (ActivityRepository, error) { //nolint: revive
+	sqlStatements := make(map[string]*sqlx.Stmt)
+
+	for queryName, query := range queriesActivity() {
+		stmt, err := db.Preparex(query)
+		if err != nil {
+			return ActivityRepository{},
+				errors.WrapErrorf(err, errors.ErrCodeUnknown, "error preparing statement %s", queryName)
+		}
+		sqlStatements[queryName] = stmt
+	}
```

Comment:
I see the point, but TBH, I don't see the reason. We want to keep the statements in memory, we don't need to close it after using it 'cause it will be closed automatically when the application shutdown, until there, it needs to be in memory. Unless someone can prove the opposite.

Reply:
@arianerocha @marianysilva, there's an aspect we should address here — though not an immediate concern. The idea behind precompiling the statement is to safeguard against potential SQL-related issues during the application bootstrap. Executing it on demand might lead to unexpected behavior, and having a precompiled statement allows us to catch any problems upfront. While this could be covered by unit tests, the upfront check offers an additional layer of security.

There's also a performance advantage as it avoids compiling the statement each time it's executed, but it comes at the cost of higher memory usage for the service. Introducing an LRUCache might help manage memory, but, as the bot rightly pointed out, it introduces added complexity.

At the moment, I'm not overly concerned—it's functioning well as is. However, I believe it's a topic worth revisiting in the future. What are your thoughts? Does this rationale resonate with you?

Reply:
@coderabbitai Open a follow-up GitHub issue for this discussion.